src/ol/geom/circle.js

```diff
@@ -186,7 +186,7 @@ ol.geom.Circle.prototype.setCenter = function(center) {
  */
 ol.geom.Circle.prototype.setCenterAndRadius = function(center, radius, opt_layout) {
   if (!center) {
-    this.setFlatCoordinates(ol.geom.GeometryLayout.XY, null);
+    this.setFlatCoordinates(opt_layout || ol.geom.GeometryLayout.XY, null);
```

Comment:
For Circle it's not needed since CIRCLE isn't part of WKT. Just thought it should be consistent and respect the sent parameters, if present.